src/hints/dict/initSquashData.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from 'zod';
+
+import { ExpectedFelt } from 'errors/primitives';
+
+import { Felt } from 'primitives/felt';
+import { isFelt } from 'primitives/segmentValue';
+import { VirtualMachine } from 'vm/virtualMachine';
+
+import { HintName } from 'hints/hintName';
+import {
+  CellRef,
+  cellRef,
+  resOperand,
+  ResOperand,
+} from 'hints/hintParamsSchema';
+
+/** Zod object to parse InitSquashData hint */
+export const initSquashDataParser = z
+  .object({
+    InitSquashData: z.object({
+      dict_accesses: resOperand,
+      ptr_diff: resOperand,
+      n_accesses: resOperand,
+      big_keys: cellRef,
+      first_key: cellRef,
+    }),
+  })
+  .transform(
+    ({
+      InitSquashData: {
+        dict_accesses,
+        ptr_diff,
+        n_accesses,
+        big_keys,
+        first_key,
+      },
+    }) => ({
+      type: HintName.InitSquashData,
+      dictAccesses: dict_accesses,
+      ptrDiff: ptr_diff,
+      nAccesses: n_accesses,
+      bigKeys: big_keys,
+      firstKey: first_key,
+    })
+  );
+
+/**
+ * InitSquashData hint
+ *
+ * Initialize the squashed dictionnary data.
+ */
+export type InitSquashData = z.infer<typeof initSquashDataParser>;
+
+/**
+ * Initialize the squashed dictionnary
+ *
+ * - Insert all accessed keys
+ * - Assert the biggest key to `bigKeys` address
+ * - Assert the smallest (first) key to `firstKey` address
+ *
+ */
+export const initSquashData = (
+  vm: VirtualMachine,
+  dictAccesses: ResOperand,
+  ptrDiff: ResOperand,
+  nAccesses: ResOperand,
+  bigKeys: CellRef,
+  firstKey: CellRef
```

Comment:
we are not going to refactor for consistency with the rest of the codebase but I think that ideally the signatures of these functions could've been `export const initSquashData(vm, {dictAccesses, ptrDiff, nAccess, bigKeys, firstKey})` which feels more aligned with the zod schema (initSquashDataInput would be a single object)

Reply:
mmh I disagree on having the inputs as a single object for hint parameters. The current zod schema flatten the hint name with its parameters, nesting them in another object would only add complexity.

For reference, the InitSquashData hint found in compilation artifacts:
```json
{
    "InitSquashData": {
        "dict_accesses": { "Deref": { "register": "FP", "offset": -4 } },
        "ptr_diff": { "Deref": { "register": "FP", "offset": 0 } },
        "n_accesses": { "Deref": { "register": "AP", "offset": -1 } },
        "big_keys": { "register": "FP", "offset": 2 },
        "first_key": { "register": "FP", "offset": 1 }
    }
}
```

Current zod schema gives
```json
{
    "type": "InitSquashData",
    "dictAccesses": {},
    "ptrDiff": {},
}
```

and your proposal would give:
```json
{
   "type": "InitSquashData",
    "initSquashDataInput": {
        "dictAccesses": {},
        "ptrDiff": {},
    }
}
```
It will reduce readability in the core logic function of the hint, either having to destruct the object in multiple variables `const {dictAccesses, ptrDiff} = initSquashDataInput`,  or having an unnecessary verbose name `initSquashDataInput.dictAccesses`...)

Reply:
fair enough :)